pkg/traceql/ast.go

```diff
@@ -721,6 +727,15 @@ func (s Static) MapKey() StaticMapKey {
 }
 
 func (s Static) Equals(o *Static) bool {
+	if s.Type == TypeNil && o.Type == TypeNil {
+		return true
+	}
+
+	// if one is nil, they are not equal
+	if s.Type == TypeNil || o.Type == TypeNil {
+		return false
+	}
+
```

Comment:
just to check my understanding, is this correct?

`{ nil = nil }` -> true
`{ span.foo = nil }` -> false
`{ nil = span.foo }` -> false
`{ nil != nil }` -> false
`{ span.food != nil }` -> false
`{ nil != span.food }` -> false

and now TypeNil only comes from the fetch layer and can't be the Nil from the query because we translate that into `OpExists`.

Reply:
Removing 
```
	if s.Type == TypeNil && o.Type == TypeNil {
		return true
	}
```

so that all comparisons with the value nil result in false. This will make all nil comparisons have consistent behavior in the fetch layer and the engine.